modules/core/src/main/java/org/locationtech/jts/algorithm/locate/IndexedPointInAreaLocator.java

```diff
@@ -102,15 +102,13 @@ public int locate(Coordinate p)
   private synchronized void createIndex() {
     if (index == null) {
       index = new IntervalIndexedGeometry(geom);
-      // no need to hold onto geom
-      geom = null;
```

Comment:
Is there a reason the geometry can't be freed?

Reply:
The reason is that final ``` geom```  field looks clearer, and the performance effect of nulling it is likely negligible.
However, it can be safely nulled, fixed in this way.